uxarray/grid.py

```diff
@@ -0,0 +1,272 @@
+"""uxarray grid module."""
+import math
+import xarray as xr
+import numpy as np
+from ast import Raise
+from logging import raiseExceptions
+
+from pathlib import PurePath
+import os
+
+
+class Grid:
+    """A class for uxarray grid object.
+
+    Examples
+    ========
+    import uxarray as ux
+    # open an exodus file with Grid object
+    mesh = ux.Grid("filename.g")
+
+    # save as ugrid file
+    mesh.saveas("outfile.ug")
+    """
+
+    # Import read/write methods from files in this folder
+    from ._exodus import read_exodus, write_exodus
+    from ._ugrid import read_ugrid, write_ugrid
+    from ._shpfile import read_shpfile
+    from ._scrip import read_scrip
+
+    def __init__(self, *args, **kwargs):
+        """Initialize grid variables, decide if loading happens via file, verts
+        or gridspec If loading from file, initialization happens via the
+        specified file.
+
+        Raises:
+            RuntimeError: File not found
+        """
```

Comment:
What are the args for this function? The possible variables below should probably be included as kwargs if they aren't always applicable.

Reply:
This is developing, and there are not that many variables now.
I'll put a placeholder with a couple of parameters to guide it.

Reply:
I'd suggest putting a #todo comment for this type of developing parts and explaining what are to be discussed/improved in the future. That would give us the opportunity to comprehensively know our parts of the code to be developed as well as  the rest of the team  to give the same thing some thinking.

Reply:
>I also think we should prioritize adding expected inputs and outputs to the doc strings as we go for clarity for everybody reading the source code.
I second this. We should populate the doc strings along with implementation. For example, "latlon", "gridspec" that are already used in the implementation need to be described/explained in the doc strings.